src/components/radio-button/radiobutton.test.js

```diff
@@ -0,0 +1,622 @@
+            <RadiobuttonGroupComponent legendWidth={width} legendInline />
+          );
+          radiobuttonGroupLegend().should("have.attr", "width", width);
+        }
+      );
+
+      it("should render RadiobuttonGroup component with children", () => {
+        CypressMountWithProviders(
+          <RadiobuttonGroupComponent>
+            <RadioButton id="radio-4" value="radio4" label="Don't Know" />
+          </RadiobuttonGroupComponent>
+        );
+        radiobuttonGroup().should("contain.text", "Don't Know");
+      });
+
+      it.each([
+        [1, "8px"],
+        [2, "16px"],
+      ])(
+        "should render RadiobuttonGroup component with legendSpacing set to %s",
+        (spacing, padding) => {
+          CypressMountWithProviders(
+            <RadiobuttonGroupComponent
+              legendSpacing={spacing}
+              legendWidth="10"
+              legendInline
+            />
+          );
+          radiobuttonGroupLegend().should("have.css", "padding-right", padding);
+        }
+      );
+
+      it.each(["top", "bottom", "left", "right"])(
+        "should render RadiobuttonGroupComponent component with tooltip positioned to the %s",
+        (position) => {
+          CypressMountWithProviders(
+            <RadiobuttonGroupComponent
+              error="Something is wrong"
+              tooltipPosition={position}
+            />
+          );
+
+          radiobuttonGroupIcon().trigger("mouseover");
+          tooltipPreview()
+            .should("have.text", "Something is wrong")
+            .should("have.attr", "data-placement", position);
+        }
+      );
+
+      it("should render RadiobuttonGroup component as a required field", () => {
+        CypressMountWithProviders(<RadiobuttonGroupComponent required />);
+
+        verifyRequiredAsteriskForLegend();
+      });
+
+      it("should render RadiobuttonGroup with radiobuttons inline", () => {
+        CypressMountWithProviders(<RadiobuttonGroupComponent inline />);
+
+        radiobutton(0)
+          .should("have.css", "margin-bottom", `12px`)
+          .should("have.css", "margin-left", "0px");
+
+        radiobutton(1)
+          .should("have.css", "margin-bottom", `12px`)
+          .should("have.css", "margin-left", `32px`);
+
+        radiobutton(2)
+          .should("have.css", "margin-bottom", `0px`)
+          .should("have.css", "margin-left", `32px`);
+      });
+
+      it.each([
+        ["inline", "399"],
+        ["inline", "400"],
+        ["above", "401"],
+      ])(
+        "should check RadiobuttonGroup legend is %s with adaptiveLabelBreakpoint %s and viewport 400",
+        (alignment, breakpoint) => {
+          cy.viewport(400, 300);
+
+          CypressMountWithProviders(
+            <RadiobuttonGroupComponent
+              adaptiveLegendBreakpoint={breakpoint}
+              adaptiveSpacingBreakpoint={breakpoint}
+            />
+          );
+
+          if (alignment === "inline") {
```

Comment:
and after the change above we will be able to avoid `IF` statement
and do like:
`radiobuttonGroupLegend().should("have.css", "float", float);`

Reply:
Updated